### PR TITLE
feat(vpn-desktop): add exit node location support

### PR DESCRIPTION
**Pull request by zaneschepke**
# Description

Add entry node selection setting to settings screen

Toggling off entry node selection resets entry node to default quick selection node

Country list loaded from storage cache on app load

Request update to country list cache on navigation to NodeLocation screen

Set Quick connect text to 'Recommended' and display bolt for recommended country

### CODE REVIEW

File: nym-vpn/ui/src-tauri/src/commands/app_data.rs [hunk 1]
```diff
@@ -97,3 +93,26 @@ pub async fn set_ui_theme(
...
+
+#[instrument(skip(data_state))]
+#[tauri::command]
+pub async fn set_entry_selector(
+    data_state: State<'_, SharedAppData>,
+    entry_selector: bool,
+) -> Result<(), CmdError> {
+    debug!("set_entry_selector");
+
+    // save the selected UI theme to disk
```
Comment (doums) [inline nym-vpn/ui/src-tauri/src/commands/app_data.rs]: un-related comment

File: nym-vpn/ui/src/pages/home/Home.tsx [hunk 2]
```diff
@@ -73,11 +73,23 @@ function Home() {
         <div className="flex flex-col justify-between">
           <NetworkModeSelect />
           <div className="py-2"></div>
-          <HopSelect
-            country={exitNodeLocation ?? QuickConnectCountry}
-            onClick={() => navigate(routes.exitNodeLocation)}
-            nodeHop="exit"
-          />
+          <div className="flex flex-col gap-6">
+            <div className="mt-3 font-semibold text-lg">Connect to</div>
```
Comment (doums) [inline nym-vpn/ui/src/pages/home/Home.tsx]: please for any copy text, use translation key

File: nym-vpn/ui/src-tauri/src/commands/connection.rs [hunk 3]
```diff
@@ -67,23 +70,37 @@ pub async fn connect(
...
+    let mut vpn_config = create_vpn_config(entry_point, exit_point);
+    if let VpnMode::TwoHop = app_state.vpn_mode {
+        info!("2-hop mode enabled");
+        vpn_config.enable_two_hop = true;
+    } else {
+        info!("5-hop mode enabled");
     }
     // vpn_config.disable_routing = true;
+    // !! release the mutex
+    drop(app_state);
```
Comment (octol) [inline nym-vpn/ui/src-tauri/src/commands/connection.rs:114]: scope ;)

**Review by octol — APPROVED** [review]

File: nym-vpn/ui/src-tauri/src/commands/connection.rs (hunk 3 above)
Comment (doums) [reply nym-vpn/ui/src-tauri/src/commands/connection.rs:114]: yea but not so simple to scope those, let's stick with drop
a refactoring would be better here, extracting logic parts into dedicated function

**Pull request merged**